lib/cyrusdb_skiplist.c

```diff
@@ -1387,8 +1359,7 @@ static int mystore(struct dbengine *db,
 
     if (localtid) {
         /* commit the store, which releases the write lock */
-        r = mycommit(db, tid);
-        if (r) return r;
```

Comment:
I like many of the simplifications in here, but I don't like this one because it's a flow break.  It changes the space after this and before the final 'return 0' from "runs if there is no error" to "never runs if there's a localtid", which makes it more complex to add additional code there.

Returning early on non-error cases makes it more complex to extend code and isn't an automatic good.